src/main/java/com/avispl/symphony/dal/avdevices/avcontrollers/planar/walldirector/WallDirectorCommunicator.java

```diff
@@ -0,0 +1,1020 @@
+                            stats.put(group.concat(WallDirectorConstant.HASH).concat(name).concat(WallDirectorConstant.CURRENT_VALUE).concat("(%)"), valueColor);
+                        }
+                        break;
+                    case GRAY_BALANCE_GAMMA:
+                        String[] grayColorValues = value.trim().split(WallDirectorConstant.SPACE_REGEX);
+                        if (grayColorValues.length != 3) {
+                            logger.error("Invalid position format when populate Color Gray Balance");
+                        }
+                        for (int index = 0; index < colors.length; index++) {
+                            String name = group.concat(WallDirectorConstant.HASH).concat("ColorGrayBalance").concat(colors[index]);
+                            String valueColor = grayColorValues[index];
+                            addAdvancedControlProperties(advancedControllableProperties, stats,
+                                    createSlider(stats, name, "1.8", "2.6", 1.8f, 2.6f, Float.valueOf(valueColor)), valueColor);
+                            stats.put(name.concat(WallDirectorConstant.CURRENT_VALUE), valueColor);
+                        }
+                        break;
+                    case CABINET_INPUT:
+                        addAdvancedControlProperties(advancedControllableProperties, stats, createDropdown(propertyName, CabinetInputEnum.toArray(), value), value);
+                        break;
+                    default:
+                        stats.put(propertyName, value);
+                        break;
+                }
+            }
+        }
+    }
+
+    /**
+     * check value is null or empty
+     *
+     * @param value input value
+     * @return value after checking
+     */
+    private String getDefaultValueForNullData(String value) {
+        return StringUtils.isNotNullOrEmpty(value) ? value.replace("\"", "") : WallDirectorConstant.NONE;
+    }
+
+    /**
+     * Adds or updates an advanced controllable property and its value in the provided lists.
+     *
+     * @param advancedControllableProperties the list of controllable properties
+     * @param stats the map of stats to update with the property value
+     * @param property the property to add or update
+     * @param value the associated value
+     */
+    private void addAdvancedControlProperties(List<AdvancedControllableProperty> advancedControllableProperties, Map<String, String> stats, AdvancedControllableProperty property, String value) {
+        if (property != null) {
+            for (AdvancedControllableProperty controllableProperty : advancedControllableProperties) {
+                if (controllableProperty.getName().equals(property.getName())) {
+                    advancedControllableProperties.remove(controllableProperty);
+                    break;
+                }
+            }
+            if (StringUtils.isNotNullOrEmpty(value)) {
+                stats.put(property.getName(), value);
```

Comment:
You may also want to update timestamp of the controllable property, when it's created/updated